honeycombio/intro-to-o11y-java
# Grouping instruction paths together (Gitpods vs local) for enhanced clarity based on user feedback

## Which problem is this PR solving?

Updating the Gitpod setup flow and instructions for Gitpod and local based on user research feedback

## Short description of the changes

1. Starting the app fails if the API key env variable isn't set, so updating the Gitpods startup script and instructions to not start the app until the user has set up the API key.
2. Restructuring the instructions a bit so the user can follow all the steps for Gitpods or local setup one after the other and only branch at the beginning
3. Adding screenshots for a couple of the trickier parts of the Gitpods setup
4. Updating dependency versions in the pom and readme for consistency

## How to verify that this has the expected result

Use the documented instructions for Gitpod and verify you are able to get data into Honeycomb
